### ISSUE TITLE
Highlights not saving

### ISSUE THREAD

**Issue by Computer-Illiterate** [luoliyan/incremental-reading#35]
Hi,
I'm currently using Anki Version 2.0.48, IR v3.8.2 on my Mac OS 10.12.6. For some reason none of my highlights will save on my IR cards, whether I'm just highlighting it, or extracting text. The highlights will show up on the card but when I come back to review them they will be gone.

I think I've seen others with macs post about this problem before but I was unsure if there has been a solution with a newer version of Anki/IR. I'm unable to update at this time because I need the Image Occlusion feature.

On an unrelated note, is there a way to make the title of my card scroll down with me as I move down through longer cards? This would be really helpful for me on cards where I pick up where I left off and naming extracted pieces of the articles.

I pretty much nothing about computers so any help would be really appreciated and please let me know if there is any other information that would be helpful. Thank you for all you do with this add on!

**Comment by luoliyan** [comment]
Thanks for reporting this. From memory, there have been two other Mac users who've run into the same issue. Unfortunately, I don't have a Mac, so it's a bit difficult for me to test. 

Can you tell me, are you able to delete text? Does the text remain deleted when you return to the card? If so, it's probably a general problem with saving the changes.

I think I have a decent plan to sort this stuff out. Over the weekend, I'll try to split-up the components of the `save` function and wrap some tests around them. GitHub has a feature that can run tests on Mac, so I'll have a go at setting that up.

In the mean time, feel free to open an issue on anything else that behaves oddly on Mac. I'd hate for this to be an add-on that's only useful for Windows/Linux/BSD folks.

**Comment by Computer-Illiterate** [comment]
I can delete the text, but it does not remain deleted when I review the card. My scroll position will save on my cards though, not sure if thats relevant.

I also just created a new anki profile and tested the highlighting and deleting features on an IR card. They appear to be saving on this other profile, so maybe it's a problem with the settings on my main profile?

**Comment by Computer-Illiterate** [comment]
Another issue I have that may be related to a general problem with saving changes; when I extract pieces of a wikipedia article the pictures will no longer appear. They will have the blue check mark symbol. The pictures show up fine on the original cards, they just disappear when extracted to new cards.

**Comment by luoliyan** [comment]
Thanks for the information. I think the problem with the images might be related to #27. I need to take another look at how images are being handled in general.

**Comment by Computer-Illiterate** [comment]
Hi luoliyan,

I just wanted to check in and see if you had a chance to look at the bug yet. Any ideas for a fix would be greatly appreciated!

Thank you

**Comment by luoliyan** [comment]
I'm still thinking about how to address this. I've been playing around with a few tests, but I haven't got much further than that.

I've got a few other changes I need to finish off - after that, I'll start working on this issue again.

Thanks for your interest.

**Comment by Computer-Illiterate** [comment]
Okay, Thanks for the update!

**Comment by Computer-Illiterate** [comment]
Hi Luoliyan,

I just wanted to give you an update so you don't waste anymore of your time with my problem.
I found the reasoning behind my highlights and deletions not saving. Some time ago when I first started using the IR add on my title of the articles was not appearing. So to try and fix this I altered the Front template of the IR3 card. Well I managed to get my title to show up and I just forgot about it.

When trying to figure out my problem I remembered that I had changed the front template, and I changed it back to the original settings and sure enough, Everything started to save!

However my title's were again not showing up, so I set the front template to
 `{{Title}}

<div class="ir-text">{{Text}}</div>
` 
Now the highlights save and my title shows up. The title now appears in the text field when I'm viewing a card though, instead of being in it's own field. If that makes sense. So since I have no experience with code, I'm sure I didn't do something right.

Anyways, turns out it was my own stupid fault the whole time, sorry for any time you have wasted trying to figure out this issue!

Thanks again for the help.
